windows/src/developer/TIKE/xml/kmw/index.html

```diff
@@ -114,7 +156,69 @@ <h2>Keyboard Test Host</h2>
           document.getElementById('install_link').href='keyman://localhost/open?direct=true&url='+location.protocol+'//'+location.host+'/kbinstall/'+nm+'-'+debugKeyboards[nm].version+'.json';
         });
         
-      //keyman.addEventListener
+      var ta1 = document.getElementById('ta1');
+      var charGrid = document.getElementById('character-grid');
+      var lastContent = null;
+      
+      if(keyman.util.isTouchDevice()) {
+        document.body.className += ' touch-device';
+      } else {
+        document.body.className += ' desktop-device';
+      }
+      
+      function removeChildNodes(node) {
+        while (node.lastChild) {
+          node.removeChild(node.lastChild);
+        }
+      }
+      
+      function addCharElements(text, code) {
+        var ebox = document.createElement('div'), echar = document.createElement('div'), ecode = document.createElement('div');
+        echar.textContent = text;
+        echar.className = 'char-char keymanweb-font';
+        ecode.textContent = code;
+        ecode.className = 'char-code';
+        ebox.appendChild(echar);
+        ebox.appendChild(ecode);
+        charGrid.appendChild(ebox);
+      }
+      
+      function logContent() {
+        if(lastContent === ta1.value) return;
+        removeChildNodes(charGrid);
+        if(ta1.value.length == 0) {
+          addCharElements('-','empty');
+        } else {
+          for(var i = 0; i < ta1.value.length; i++) {
+            //
+            var code = ta1.value.charCodeAt(i);
+            var text = ta1.value.charAt(i);
+            var slice = 4;
+            // Test for SMP
+            if(code >= 0xD800 && code < 0xDC00) {
+              if(i < ta1.value.length) {
+                var code2 = ta1.value.charCodeAt(i+1);
+                if(code2 >= 0xDC00 && code < 0xE000) {
+                  code = (code - 0xD800) * 0x400 + (code2 - 0xDC00) + 0x10000;
+                  text += ta1.value.charAt(i+1);
+                  slice = 6;
+                  i++;
+                }
+              }
+            }
+            addCharElements(text, ('000000'+(code).toString(16)).slice(-slice));
+          }
+        }
+        console.log(ta1.value);
+        lastContent = ta1.value;
+      }
+
+      logContent();
+      window.setInterval(logContent, 100);
+        
+      /* TODO: once KeymanWeb supports oninput signalling
```

Comment:
Maybe add reference to issue #571

Reply:
I'll consider your comment sufficient reference -- this change is now shown in the issue.